### PR TITLE
feat(awscdk): add AWS Lambda Extension development support

### LINKED ISSUE #1646: (awscdk): lambda extensions
I've been working on an [AWS Lambda Extension](https://aws.amazon.com/blogs/compute/building-extensions-for-aws-lambda-in-preview/) Construct for CDK projects, and getting started was more challenging than I'd have liked. I'd like Projen to support developing AWS Lambda Extensions, possibly creating them as Lambda Layers using Projen's auto-discover & bundling support.

### PR DESCRIPTION
**Pull request by misterjoshua**
Adds AWS Lambda Extension development support to AWS CDK projects.

Fixes #1646

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by misterjoshua** [comment]
If supporting TS-based Lambda Extensions is too specific, we could simplify extension bundling by keeping just the bundler `outfile` and `outdir` changes.

File: src/awscdk/lambda-extension.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    const bundle = bundler.addBundle(options.entrypoint, {
+      platform: bundlerRuntime.esbuildPlatform,
+      target: bundlerRuntime.esbuildTarget,
+      externals: ["aws-sdk"],
+      outfile: `extensions/${name}`,
+      ...options.bundlingOptions,
+    });
+
+    bundle.bundleTask.exec(`chmod +x ${bundle.outfile}`);
```
Comment (Chriscbr) [inline src/awscdk/lambda-extension.ts:126]: Why is this needed? Can we add a comment?

Would it make sense to add this as an option to addBundle instead, like `executable` or `markExecutable`?

File: docs/awscdk.md [hunk 2]
```diff
@@ -101,6 +101,112 @@ new awscdk.LambdaFunction(p, {
...
+- Create a file in your project's source tree called
+  `my-extension.lambda-extension.ts`
+- Run `npx projen`
+- Projen will automatically discover this file, generating an AWS Lambda Layer
+  Version named `MyExtensionLayerVersion` in a file named
+  `my-extension-layer-version.ts`.
+- Now you can instantiate `MyExtensionLayerVersion` and add it to your Lambda
+  functions.
+
+Example extension:
```
Comment (Chriscbr) [inline docs/awscdk.md:126]: Can you add a blurb about what this extension does? Without familiarity with lambda extensions, I don't think it's super clear what it does / where I would insert my functionality.

Comment (Chriscbr) [inline docs/awscdk.md:126]: Let's also add a link to this samples repo for good measure: https://github.com/aws-samples/aws-lambda-extensions

**Review by Chriscbr — CHANGES_REQUESTED** [review]
This looks great @misterjoshua ! I just added a few small nitpicks and questions 🙂

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**